File: sales_by_match.js
"use strict"

const fs = require("fs")

process.stdin.resume()
process.stdin.setEncoding("utf-8")

let inputString = ""
let currentLine = 0

process.stdin.on("data", (inputStdin) => {
  inputString += inputStdin
})

process.stdin.on("end", (_) => {
  inputString = inputString
    .replace(/\s*$/, "")
    .split("\n")
    .map((str) => str.replace(/\s*$/, ""))

  main()
})

function readLine() {
  return inputString[currentLine++]
}

// Complete the sockMerchant function below.
function sockMerchant(n, ar) {
  let pairs = 0
  let socks = {}

  ar.forEach((sock) => {
    if (socks[sock]) {
      socks[sock]++
    } else {
      socks[sock] = 1
    }
    if (socks[sock] == 2) {
      socks[sock] = 0
      pairs++
    }
  })

  return pairs
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH)

  const n = parseInt(readLine(), 10)

  const ar = readLine()
    .split(" ")
    .map((arTemp) => parseInt(arTemp, 10))

  let result = sockMerchant(n, ar)

  ws.write(result + "\n")

  ws.end()
}

File: sales_by_match.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the sockMerchant function below.
def sockMerchant(n, ar):
    pairs = 0
    socks = {}

    for sock in ar:
        if sock in socks:
            socks[sock] += 1
        else:
            socks[sock] = 1
        if socks[sock] == 2:
            socks[sock] = 0
            pairs += 1

    return pairs


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    ar = list(map(int, input().rstrip().split()))

    result = sockMerchant(n, ar)

    fptr.write(str(result) + '\n')

    fptr.close()
